modules/oe_newsroom_newsletter/tests/src/Functional/SubscribeMultipleNewslettersTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_newsroom_newsletter\Functional;
+
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\oe_newsroom_newsletter\Traits\OeNewsroomNewsletterTrait;
+use Drupal\user\Entity\Role;
+
+/**
+ * Test the subscription to multiple newsletters.
+ *
+ * @group oe_newsroom_newsletter
+ */
+class SubscribeMultipleNewslettersTest extends BrowserTestBase {
+
+  use OeNewsroomNewsletterTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'oe_newsroom_newsletter_mock',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->setApiPrivateKey();
+    $this->configureNewsroom();
+    $this->configureNewsletter();
+    $this->grantPermissions(Role::load(Role::ANONYMOUS_ID), [
+      'subscribe to newsletter',
+      'unsubscribe from newsletter',
+    ]);
+    $this->createNewsletterPages(TRUE);
+  }
+
+  /**
+   * Test if a user can subscribe to multiple newsletters.
+   *
+   * @group oe_newsroom_newsletter
+   */
+  public function testSubscribeMultipleNewsletters(): void {
+    $assertSession = $this->assertSession();
+    $session = $this->getSession();
+    $page = $session->getPage();
+
+    // Subscribe multiple Newsletters.
+    $this->drupalGet($this->subscribePath);
+    $assertSession->pageTextContains('Subscribe to newsletter');
+    $assertSession->pageTextContains('This is the introduction text.');
+    $page->fillField('Your e-mail', 'mail@example.com');
+    $assertSession->pageTextContains('Newsletter lists');
+    $assertSession->pageTextContains('Please select which newsletter list interests you.');
+    $page->checkField('By checking this box, I confirm that I want to register for this service, and I agree with the privacy statement');
+    $page->pressButton('Subscribe');
+    $assertSession->pageTextContains('Newsletter lists field is required.');
+    $page->checkField('Newsletter collection 1');
+    $page->checkField('Newsletter 2');
```

Comment:
Select only one, to test that 1 is required and not 2.

Reply:
Here, by configuration, the collection has more sv_ids while the second has only a single one.

Reply:
What I meant is that we need to test that the field is required, and 1 value is required, so we need to run at least one test that select one value only.